libraries/plugins/es_objects/es_objects.cpp

```diff
@@ -190,181 +187,36 @@ void es_objects_plugin_impl::remove_from_database( object_id_type id, std::strin
    }
 
-   prepare = graphene::utilities::createBulk(bulk_header, std::move(data));
-   std::move(prepare.begin(), prepare.end(), std::back_inserter(bulk));
-   prepare.clear();
-}
-
-void es_objects_plugin_impl::prepare_account(const account_object& account_object)
-{
-   account_struct acct;
-   acct.object_id = account_object.id;
-   acct.block_time = block_time;
-   acct.block_number = block_number;
-   acct.membership_expiration_date = account_object.membership_expiration_date;
-   acct.registrar = account_object.registrar;
-   acct.referrer = account_object.referrer;
-   acct.lifetime_referrer = account_object.lifetime_referrer;
-   acct.network_fee_percentage = account_object.network_fee_percentage;
-   acct.lifetime_referrer_fee_percentage = account_object.lifetime_referrer_fee_percentage;
-   acct.referrer_rewards_percentage = account_object.referrer_rewards_percentage;
-   acct.name = account_object.name;
-   acct.owner_account_auths = fc::json::to_string(account_object.owner.account_auths);
-   acct.owner_key_auths = fc::json::to_string(account_object.owner.key_auths);
-   acct.owner_address_auths = fc::json::to_string(account_object.owner.address_auths);
-   acct.active_account_auths = fc::json::to_string(account_object.active.account_auths);
-   acct.active_key_auths = fc::json::to_string(account_object.active.key_auths);
-   acct.active_address_auths = fc::json::to_string(account_object.active.address_auths);
-   acct.voting_account = account_object.options.voting_account;
-   acct.votes = fc::json::to_string(account_object.options.votes);
-
-   std::string data = fc::json::to_string(acct);
-
-   fc::mutable_variant_object bulk_header;
-   bulk_header["_index"] = _es_objects_index_prefix + "account";
-   bulk_header["_type"] = "data";
-   if(_es_objects_keep_only_current)
-   {
-      bulk_header["_id"] = string(acct.object_id);
-   }
+   auto blockchain_object_string = fc::json::to_string(blockchain_object, fc::json::legacy_generator);
+   variant blockchain_object_variant = fc::json::from_string(blockchain_object_string);
+   fc::mutable_variant_object o;
 
-   prepare = graphene::utilities::createBulk(bulk_header, std::move(data));
-   std::move(prepare.begin(), prepare.end(), std::back_inserter(bulk));
-   prepare.clear();
-}
+   adaptor_struct adaptor;
+   auto adapted = adaptor.adapt(blockchain_object_variant.get_object());
 
-void es_objects_plugin_impl::prepare_asset(const asset_object& asset_object)
-{
-   asset_struct asset;
-   asset.object_id = asset_object.id;
-   asset.block_time = block_time;
-   asset.block_number = block_number;
-   asset.symbol = asset_object.symbol;
-   asset.issuer = asset_object.issuer;
-   asset.is_market_issued = asset_object.is_market_issued();
-   asset.dynamic_asset_data_id = asset_object.dynamic_asset_data_id;
-   asset.bitasset_data_id = asset_object.bitasset_data_id;
-
-   std::string data = fc::json::to_string(asset);
+   fc::from_variant(adapted, o, FC_PACK_MAX_DEPTH);
```

Comment:
Similarly, it would be more efficient to have `adaptor_struct::adapt` return a `mutable_variant_object` directly, so you don't have to copy stuff afterwards.